specs/FamilySafety.md

```diff
@@ -0,0 +1,82 @@
+Please see https://www.microsoft.com/en-us/microsoft-365/family-safety for details on Family Safety. 
+
+# Examples
+## WinRT and .NET   
+```c#
+void CreateEnvrionmentWithOption()
+{
+    CoreWebView2EnvironmentOptions options = new CoreWebView2EnvironmentOptions();
+    options.IsFamilySafetyEnabled = true;
+    options.SetFamilySafetySoftByPassList(1, "appassets.example/AppStartPage.html");
+    auto environment = await CoreWebView2Environment.CreateAsync(BrowserExecutableFolder, UserDataFolder, options);
+}
+```
+## Win32 C++
+```cpp
+HRESULT InitializeWebView()
+{
+    // Enable the Family Safety feature upon webview environment creation complete
+    auto options = Microsoft::WRL::Make<CoreWebView2StagingEnvironmentOptions>();
+    Microsoft::WRL::ComPtr<ICoreWebView2EnvironmentOptions3> optionsStaging3;
+    if (options.As(&optionsStaging3) == S_OK)
+    {
+        optionsStaging3->put_IsFamilySafetyEnabled(TRUE);
+
+        const WCHAR* byPassLists[1] = {L"appassets.example/AppStartPage.html"};
+        optionsStaging3->SetFamilySafetySoftByPassList(1, byPassLists);
+    }
+
+    // CreateCoreWebView2EnvironmentWithOptions
+    HRESULT hr = CreateCoreWebView2EnvironmentWithOptions("", nullptr, options.Get(),
+    Callback<ICoreWebView2CreateCoreWebView2EnvironmentCompletedHandler>(
+        this, &AppWindow::OnCreateEnvironmentCompleted)
+        .Get());
+}
+```
+
+# API Details    
+```
+interface ICoreWebView2EnvironmentOptions3;
+
+/// Additional options used to create WebView2 Environment.
+[uuid(D0965AC5-11EB-4A49-AA1A-C8E9898F80AF), object, pointer_default(unique)]
+interface ICoreWebView2EnvironmentOptions3 : IUnknown {
+  /// When `IsFamilySafetyEnabled` is `TRUE` WebView2 provides the same Family Safety functionality as the Edge browser for child accounts:
+  /// Activity Reporting, Web Filtering and SafeSearch. Please see https://www.microsoft.com/en-us/microsoft-365/family-safety for details.
+  /// `IsFamilySafetyEnabled` property is to enable/disable family safety feature.
+  /// It is `FALSE` by default.
+  [propget] HRESULT IsFamilySafetyEnabled([out, retval] BOOL* value);
+  /// Sets the `IsFamilySafetyEnabled` property.
+  [propput] HRESULT IsFamilySafetyEnabled([in] BOOL value);
+
+  /// When Family Safety feature is enabled, provide ability to modify the soft by pass list so app remain functioning in BlockAll Mode,
+  /// parents can still override the URL if they want the app blocked.
+  /// \snippet AppWindow.cpp CoreWebView2FamilySafety
+  HRESULT GetFamilySafetySoftByPassList([out] UINT32* uriCounts, [out] LPWSTR** lists);
```

Comment:
The soft bypass list is not described anywhere here. Maybe we can name this something like 
 * `FamilySafetyAllowedUris`

I was also thinking of the following but I think allowed URIs is maybe the easiest to understand?
 * `FamilySafetyBypassUris`
 * `FamilySafetyUriExceptions`

Do you have other ideas?

If there is more technical documentation on family safety that describes the names of its settings and it has some kind of allowed list of URIs maybe we can reuse names from that?

Reply:
FamilySafetyAllowedUris is pretty intuitive. but FamilySafetyBypassUris sounds more like an exception? I will go with FamilySafetyAllowedUris first and open to change after official review